### PR TITLE
mrc-2247 Endpoint for queue status

### PR DESCRIPTION
**Pull request by EmmaLRussell**
Adds an API endpoint which just fetches through the queue status from orderly server. 
As requested by Rob, the endpoint is secured to authenticated users, but no particular permissions required.

### CODE REVIEW

File: docs/spec/QueueStatus.schema.json [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+          "key": {"type": "string"},
+          "status": {
+            "enum": ["queued", "running", "success", "error", "orphan", "interrupted", "redirect", "missing"]
+          },
+          "name": {"type": "string"},
+          "version": {
+            "type": [ "string", "null" ]
+          }
+        },
+        "required": ["key", "status", "name", "version"],
```
Comment (mwoodbri) [inline docs/spec/QueueStatus.schema.json:20]: Presumably this spec is inherited from orderly.server but `version` being `required` doesn't seem consistent with `spec.md` or the tests below

File: docs/spec/spec.md [hunk 2]
```diff
@@ -565,3 +565,27 @@ Required permissions: `reports.run`.
...
+
+Gets information on the current state of the orderly queue.
+
+Required permissions: none, available to all authenticated users
+
+Schema: [`QueueStatus.schema.json`](QueueStatus.schema.json)
+
+### Example
+```json
+[
```
Comment (mwoodbri) [inline docs/spec/spec.md:579]: Integration test below and schema above suggest that this should be wrapped with `{tasks: [ ... ]}`?

**Review by mwoodbri — CHANGES_REQUESTED** [review]
All looks good - apart perhaps from documentation?

**Comment by EmmaLRussell** [comment]
> All looks good - apart perhaps from documentation?

Agreed, this is a bit off in the orderly server docs which I mindlessly copied it from!

**Review by mwoodbri — APPROVED** [review]

**Review by Nicolas-Dolan — COMMENTED** [review]
Looks good. I have no changes to suggest.

**Pull request merged**